src/pyprojroot/root.py

```diff
@@ -15,13 +15,9 @@
 )
 
 
-def as_start_path(start: Union[None, _PathType]) -> Path:
-    if start is None:
-        return Path.cwd()
-    if not isinstance(start, Path):
-        start = Path(start)
-    # TODO: consider `start = start.resolve()`
-    return start
+def as_start_path(start: Optional[_PathType]) -> Path:
+    """A pathlib.Path object based on the common working directory or the optional input provided."""
+    return Path.cwd() if start is None else Path(start)
```

Comment:
I'm pretty sure that we need to do something like `resolve` or `expanduser`. So the TODO still stands.

Reply:
Hi James,

I was wondering what the usecase was.  The Path returned can have the caller call resolve().  Is there a reason for doing this step at this time?

Reply:
I'm also realizing I wrote that it is using the 'common working directory -- but its the 'current working directory'.  Would also need to fix that depending on how this conversation goes :).   I'm interested to understand the resolve aspect though.

Reply:
What happens if you call `find_root(criterion, start=".")` or `find_root(criterion, start="~/project")`?

Reply:
I asked the above question by creating a script called examination and placed it here:

`/usr/local/bin/python3 /Users/jasonbrackman/PycharmProjects/pyprojroot/tests/examination.py `

1. I had to convert the tilde (is that what you expect?)
2. I noticed that the projects gets clipped, so added a third case.  I'm assuming this is not what you expect?

Demo Code:

```
import os
from pathlib import Path
from src.pyprojroot import has_dir, find_root

criteria = has_dir(Path('/tmp'))

test01 = find_root(criteria, start='.')
test02 = find_root(criteria, start=r'~/projects'.replace("~", os.path.expanduser('~')))
test03 = find_root(criteria, start=r'~/projects/weird'.replace("~", os.path.expanduser('~')))

for test in (test01, test02, test03):
    print(f'Returned: {test!s:30} >> Resolved: {test.resolve()!s}')
```

Resulted in:
```
Returned: .                              >> Resolved: /Users/jasonbrackman/PycharmProjects/pyprojroot/tests
Returned: /Users/jasonbrackman           >> Resolved: /Users/jasonbrackman
Returned: /Users/jasonbrackman/projects  >> Resolved: /Users/jasonbrackman/projects
```

Reply:
The answer will of course depend on the criteria, but I don't think it should the user to expand or resolve the input in advance. Otherwise we would also require the user to pass a Path object rather than anything PathLike.

The biggest issue is with `start="."` since the current implementation will not look any further up the heirarchy than `"."`. For "~", it's unlikely that anyone will look for a root that is in a parent of the "~" directory.

Reply:
E.g. 

```
/user/projects/
 - big_analysis/
    - .here  <-- marker file
    - notebooks/
        - Untitled.ipynb
```

Using v0.3.0, calling `find_root(".here", start=".")` from the `notebooks` directory will raise `RuntimeError` rather than `"/user/projects/big_analysis"`.

Reply:
Updated the review to take advantage of .expanduser() before return the path.  Here is my input/output:

```
import os
from src.pyprojroot import find_root

# Set the working directory to start from
os.chdir(r'/Users/jasonbrackman/projects/big_analysis')


test01 = find_root('.here', start='.')
test02 = find_root('.here', start=r'~/projects')

for test in (test01, test02):
    print(f'Returned: {test!s:30} >> Resolved: {test.resolve()!s}')
```

Produces:

```
Returned: .                              >> Resolved: /Users/jasonbrackman/projects/big_analysis
Returned: /Users/jasonbrackman           >> Resolved: /Users/jasonbrackman
```